### ISSUE TITLE
Docusaurus cleanup - Contribution related content

### ISSUE THREAD

**Issue by ildikov** [magma/magma#12707]
### Your Environment

- **Version:** All versions
- **Affected Component:** Docusaurus

### Describe the Issue

The guidelines and pointers of how to contribute to the Magma project have been moved to a separate, standalone document called the Contributor Guide: https://github.com/magma/magma/wiki/Contributor-Guide

All versions of the Docusaurus needs to be checked and cleaned up to remove all contribution related content and replace it with pointers to the relevant sections of the Contributor Guide: https://github.com/magma/magma/tree/master/docs/docusaurus/versioned_docs

**Comment by lucasgonze** [comment]
One issue with the Docusaurus is that there should be a top-level nav item pointing to the Contributor Guide. Otherwise contributors will be frustrated finding their way back and forth between these.

**Comment by ildikov** [comment]
@lucasgonze This issue is targeting the cleanup of the current and previous versions of the Docusaurus, which is Magma's project documentation.

#12708 is targeting to remove duplicated content from the community repo, which is completely independent from the Docusaurus. Once the cleanup work is done and we ensured that all content in that repo is covered at the appropriate place the TSC will advise on the next steps with that repo, which will likely be a process to archive it.

Does this answer your question?

**Comment by Neudrino** [comment]
> One issue with the Docusaurus is that there should be a top-level nav item pointing to the Contributor Guide. Otherwise contributors will be frustrated finding their way back and forth between these.

Feel free to implement the suggested change. :slightly_smiling_face:

**Comment by lucasgonze** [comment]
> Feel free to implement the suggested change.

I have spun off a new issue for nav bar changes:
[Update docusaurus top nav](https://github.com/magma/magma/issues/13386)

**Comment by lucasgonze** [comment]
(per @Neudrino's point at https://github.com/magma/magma/issues/12708#issuecomment-1191622868, I have moved this comment from 12708).

- [ ] https://docs.magmacore.org/docs/contributing/contribute_proposals is a dup of https://github.com/magma/magma/wiki/Adding-and-Tracking-Proposals

- [ ] The link towards https://docs.magmacore.org/docs/contributing/contribute_onboarding in https://docs.magmacore.org/docs/basics/prerequisites should be changed to https://github.com/magma/magma/wiki

- [ ] There are three "opinionated developer setup" pages. They should be refactored and should move to the Contributor's Guide. These pages are  
  - [ ] https://github.com/magma/magma/wiki/Opinionated-set-up-of-development-environment-for-MacOS
  - [ ] https://docs.magmacore.org/docs/basics/quick_start_guide
  - [ ] https://docs.magmacore.org/docs/basics/prerequisites

- [ ] The Documentation Overview in Docusaurus (https://docs.magmacore.org/docs/docs/docs_overview) should move to the Contributor's Guide because it is for contributors working on documentation.

- [ ] The pages in Docusaurus under "Contribute > Contribute" (in the left nav) are duplicates of content in the Contributor's Guide and need to be eliminated: 

1. https://docs.magmacore.org/docs/contributing/contribute_onboarding
2. https://docs.magmacore.org/docs/contributing/contribute_workflow
3. https://docs.magmacore.org/docs/contributing/contribute_vscode
4. https://docs.magmacore.org/docs/contributing/contribute_conventions
5. https://docs.magmacore.org/docs/contributing/contribute_id_mappings
6. https://docs.magmacore.org/docs/contributing/contribute_ci_checks

**Comment by ildikov** [comment]
The 'Contribute' section in the Docusaurus appeared in the 1.4.x release (https://docs.magmacore.org/docs/1.4.x/contributing/contribute_onboarding). As @lucasgonze summarized, the best approach forward is probably to point to the root of the Contributor Guide (https://github.com/magma/magma/wiki/Contributor-Guide) and remove the whole section from the Docusaurus. That needs to get done in every release starting from 1.4.x.

We only need to double check if we have everything covered in the new location that was described in the 'Contribute' section.

**Comment by lucasgonze** [comment]
> We only need to double check if we have everything covered in the new location that was described in the 'Contribute' section.

Done. I have done a paragraph-by-paragraph comparison. There are some small differences but all the content has been moved over.

@ildikov ^

**Comment by ildikov** [comment]
@lucasgonze Great news, thank you for going through it! Do I understand it correctly that the tasks listed above in this issue still need to be finished before closing it?

**Comment by lucasgonze** [comment]
@ildikov :) You understand correctly. 

I think I will soon be able to check off everything but 'three "opinionated developer setup" pages'.

**Comment by lucasgonze** [comment]
When https://github.com/magma/magma/pull/14235 is merged this ticket can be closed.

**Comment by Neudrino** [comment]
> When #14235 is merged this ticket can be closed.

There is a Github functionallity for that. See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue. (I already linked it now.)
